src/lava/lib/optimization/solvers/generic/builder.py

```diff
@@ -240,7 +241,8 @@ def constructor(self, proc):
                     shape=proc.variable_assignment.shape)
                 macrostate_reader.cost_convergence_check = c
                 variables.local_cost.connect(macrostate_reader.cost_in)
-                proc.vars.optimality.alias(macrostate_reader.min_cost)
+                proc.vars.optimality.alias(macrostate_reader.cost)
+                proc.vars.optimum.alias(macrostate_reader.min_cost)

```

Comment:
I think these names for the aliased vars can be a bit misleading, can we change them to something like `aliased_cost` and `aliased_min_cost`?

Reply:
Totally agree on hiding the abstraction from the user. 

From the developer perspective though, I would go for more informative names, like `cost_tracker` and `min_cost_tracker`, just to clarify the intended meaning of each variable. `optimum` for example can be quite ambiguous and can be easily mistaken for the "optimal variable assignment".

That being said, for me we can also merge as it is.